_studio/shared/src/libmfx_core_hw.cpp

```diff
@@ -0,0 +1,114 @@
+#include "libmfx_core_hw.h"
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <dirent.h>
+#include <string>
+#include <list>
+#include <fstream>
+#include <iomanip>
+
+const std::string PCI_DIR = "/sys/bus/pci/devices";
+constexpr mfxU32 PCI_DISPLAY_CONTROLLER_CLASS = 0x03;
+constexpr mfxU32 CLASS_ID_OFFSET = 16;
+constexpr mfxU32 INTEL_VENDOR_ID = 0x8086;
+
+int mfx_dir_filter(const struct dirent* dir_ent)
```

Comment:
I doubt we will need this function for Windows. Can it be moved closer to Linux part, and optimally - made via lambda? or at least declared static?

Reply:
`dirent` has a C interface, therefore I don't think we can pass lambdas to it.